app/benchmarks/benchmark_pfb_test.go

```diff
@@ -260,79 +271,61 @@ func BenchmarkProcessProposal_PFB_Half_Second(b *testing.B) {
 func benchmarkProcessProposalPFBHalfSecond(b *testing.B, count, size int) {
 	testApp, rawTxs := generatePayForBlobTransactions(b, count, size)
 
-	targetTimeLowerBound := 0.499
-	targetTimeUpperBound := 0.511
-
-	start := 0
-	end := count
-	segment := end - start
-	maxIterations := 100000
-	iterations := 0
-	for {
-		iterations++
-		if iterations >= maxIterations {
-			b.Errorf("Maximum iterations reached without achieving target processing time")
-			break
-		}
-		if segment == 1 {
-			break
-		}
+	targetTimeLowerBound := time.Millisecond * 499
+	targetTimeUpperBound := time.Millisecond * 511
+
+	start := 1
+	end := len(rawTxs)
+	var bestTime time.Duration
+	found := false
+
+	var processProposalReq types.RequestProcessProposal
 
-		prepareProposalRequest := types.RequestPrepareProposal{
-			Txs:    rawTxs[start:end],
-			Height: 10,
+	for start <= end {
```

Comment:
I reworked this test to use a binary search based approach, as I found that when I was running these locally and trying to debug with the upgrade to v0.50 that in some instances the txNum would keep oscillating back and forth and wouldn't be able to hit within the bounds. 

It seems to work nice enough now. The benchmark doesn't take quite as long to run, in some random instances it can hit the case where it won't find a ProcessProp time at the target but at least it does not run indefinitely, and running the bench again will normally succeed.
It could probably be further reworked to never hit this, but I didn't want to introduce too many changes. You will never get a deterministic ProcessProp time from what I can see.